src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -555,6 +555,45 @@ public void Dispose()
                 }
             }
         }
+
+        private Assembly? ResolveSatelliteAssembly(AssemblyName assemblyName)
+        {
+            // Called by native runtime when CultureName is not empty
+            Debug.Assert(assemblyName.CultureName?.Length > 0);
+
+            string cultureName = assemblyName.CultureName!;
+
+            string satelliteSuffix = ".resources";
+
+            if (assemblyName.Name == null || !assemblyName.Name.EndsWith(satelliteSuffix))
```

Comment:
EndsWith is "current culture" by default. Pass in `StringComparison.InvariantCulture` ?

Reply:
should it be OrdinalIgnoreCase even?

Reply:
`StringComparison.InvariantCulture` or `StringComparison.Ordinal` would seem interchangeable here. I guess I would choose `StringComparison.Ordinal` because it could be  faster.

`IgnoreCase`...  In normal code flow, `GetSatellite()` appends the lower case ".resources".  Tooling will create lower case ".resources". Ignoring case would allow customer to have unconventional filename if they manually tried to load the satellite. I am not sure we want to support that. It would be a poorly supported scenario.